BookPlayer/Library/PlaylistViewController.swift

```diff
@@ -13,7 +13,22 @@ class PlaylistViewController: ItemListViewController {
     var playlist: Playlist!
 
     override var items: [LibraryItem] {
-        return self.playlist.books?.array as? [LibraryItem] ?? []
+        let items: [LibraryItem] = self.playlist.books?.array as? [LibraryItem] ?? []
+
+        if let searchText = searchController.searchBar.text,
```

Comment:
we could write this part with a `guard` statement and another minor improvement I think:

```swift
let items: [LibraryItem] = self.playlist.books?.array as? [LibraryItem] ?? []

guard let searchText = searchController.searchBar.text,
  !searchText.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty else {
  return items
}

return items.filter { item in
  if let book = item as? Book {
    return book.title.contains(searchText) || book.author.contains(searchText)
  }

  // since all LibraryItem have title, we can skip the check of playlists
  // unless we plan on searching the string among the playlist contents, which could make sense 🤔
  return item.title.contains(searchText)
}
```

Reply:
oh wait, this block only applies to playlists, and since we don't support nested playlists, the filter block could just be `item.title.contains(searchText)`